### DISCUSSION TITLE
Fill unrecognized options

### DISCUSSION THREAD

**Discussion by leviska** [tokio-rs/prost#715] [Q&A] [answered]
I'm writing custom `service_generator` for http backend, and have this code sample

```rust
fn method_config(method: &Method) -> Option<TokenStream> {
  if !method.options.uninterpreted_option.is_empty() {
    panic!("{:#?}", method.options.uninterpreted_option);
  }
  // <...>
}
```

And I have this proto 

```proto
//<...>
rpc GetSources(GetSourcesRequest) returns (GetSourcesResponse) {
  option (google.api.http) = {
    get : "/sources"
  };
}
```

And compilation doesn't fail, meaning the `uninterpreted_options` are empty
In the docs there is a comment about something related:

> A message representing a option the parser does not recognize. This only appears in options protos created by the compiler::Parser class. DescriptorPool resolves these when building Descriptor objects. Therefore, options protos in descriptor objects (e.g. returned by Descriptor::options(), or produced by Descriptor::CopyTo()) will never have UninterpretedOptions in them.

But it seems outdated, because I can't find `compiler::Parser` at all. 

I've tested `deprecated` option, and it works fine, so I really don't understand why can't `prost` fill this option too.

I really want to parse this option, but as of now can't figure out how to do it. Is there a way, can I help contribute to prost with something to make this happen?

**Comment by LucioFranco** [comment, marked as answer]
The type is generated from the google protobuf types here https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/descriptor.proto#L741 and probably refers to the C++ implementation. We just copy the comments.  Prost just uses `protoc` so maybe there is a flag you can set to get that. Prost doesn't do much special here than other implementations as for getting the data from the file descriptor set.

**Reply by leviska** [reply]
oof, that's sad, but I'll look into it, thanks :)

**Reply by leviska** [reply]
As I understand, this options are stored in extensions: https://stackoverflow.com/questions/41165197/how-to-get-the-method-options-from-protobuf-descriptor

I've checked the raw output of the protoc, and there are some strings with my http paths, but in the resulting `FileDescriptorProto` there are none. Any thoughts about how can I integrate extensions in Rust?

**Reply by leviska** [reply]
Ah, I see it here https://github.com/tokio-rs/prost/issues/674
